rllib/core/loss_and_optim/rl_loss_and_optim.py

```diff
@@ -0,0 +1,75 @@
+import abc
+from typing import Mapping, Any, List
+
+from ray.rllib.core.rl_module import RLModule
+from ray.rllib.core.rl_module.marl_module import MultiAgentRLModule
+
+class RLLossAndOptim(abc.ABC):
+    """Base class for defining a loss function and optimizer for a RLModule.
+    
+    Args:
+        rl_module: The RLModule that will be optimized.
+        config: The configuration for the optimizer.
+
+    Abstract Methods:
+        compute_loss: computing a loss to optimize rl_module over.
```

Comment:
`compute_loss` does not imply that the input is not `sample_batch` and is actually the output of the model. We need to name so that the caller knows that this module won't handle the forward pass for them. Something like `compute_loss_on_module_output` may be a better name imo.

Reply:
we can call it `compute_update` or `compute_update_terms`

Reply:
compute_optimization_vars

Reply:
this implies all computing all the terms needed for an update.

Reply:
then we can change `compute_gradients_and_apply` to `optimize`